src/bitcoin-cli.cpp

```diff
@@ -748,11 +753,12 @@ static UniValue ConnectAndCallRPC(BaseRequestHandler* rh, const std::string& str
                 }
             }
             break; // Connection succeeded, no need to retry.
-        } catch (const CConnectionFailed&) {
-            if (fWait) {
-                UninterruptibleSleep(std::chrono::milliseconds{1000});
+        } catch (const CConnectionFailed& e) {
+            const int64_t now = GetTime<std::chrono::seconds>().count();
+            if (fWait && (timeout == 0 || now < deadline)) {
```

Comment:
Maybe `timeout <= 0` - zero or negative means no timeout.

Reply:
Added check using `assert_greater_than_or_equal` and had to import `time` but that should work ok.